File: 3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        freq = [0] * 26
        for ch in word:
            freq[ord(ch) - 97] += 1
        freq.sort()  # ascending

        min_pushes = 0
        push_cnt = 0
        for i in range(25, -1, -1):
            curr_idx = 25 - i
            if curr_idx % 8 == 0:
                push_cnt += 1
            min_pushes += freq[i] * push_cnt
        return min_pushes

File: 3016-minimum-number-of-pushes-to-type-word-ii/3016-minimum-number-of-pushes-to-type-word-ii.ts
function minimumPushes(word: string): number {
    const freq = new Array(26).fill(0);
    for (const ch of word) {
        freq[ch.charCodeAt(0) - 97]++;
    }
    freq.sort((a, b) => a - b); // ascending

    let minPushes = 0;
    for (let i = 25, pushCnt = 0; i >= 0; i--) {
        const currIdx = 25 - i;
        if (currIdx % 8 === 0) pushCnt++;
        minPushes += freq[i] * pushCnt;
    }
    return minPushes;
}